File: web/src/api/partTimePosition.js
import service from "@/utils/request";

export const getPartTimePositionList = (params) => {
  return service({
    url: "/partTimePosition/getList",
    method: "get",
    params,
  });
};
export const AddPartTimePosition = (data) => {
  return service({
    url: "/partTimePosition/add",
    method: "post",
    data,
  });
};
export const deletePartTimePosition = (params) => {
  return service({
    url: "/partTimePosition/delete",
    method: "delete",
    params,
  });
};
export const getJobList = (params) => {
  return service({
    url: "/partTimePosition/getJobList",
    method: "get",
    params,
  });
};
export const createJob = (data) => {
  return service({
    url: "/partTimePosition/createJob",
    method: "post",
    data,
  });
};
export const updateJob = (data) => {
  return service({
    url: "/partTimePosition/updateJob",
    method: "post",
    data,
  });
};
export const deleteJob = (params) => {
  return service({
    url: "/partTimePosition/deleteJob",
    method: "delete",
    params,
  });
};

File: server/router/hospitalManage/partTimePosition.go
package hospitalManage

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type PartTimePositionRouter struct{}

func (p *PartTimePositionRouter) InitPartTimePositionRouter(Router *gin.RouterGroup, RouterPub *gin.RouterGroup) {
	PartTiemPositionRouter := Router.Group("partTimePosition").Use(middleware.OperationRecord())
	PartTimePositionApi := v1.ApiGroupApp.HospitalManageApiGroup.ManagePartTimePositionApi
	{
		PartTiemPositionRouter.GET("getList", PartTimePositionApi.GetPartTimePositionList)
		PartTiemPositionRouter.POST("add", PartTimePositionApi.AddPartTimePosition)
		PartTiemPositionRouter.DELETE("delete", PartTimePositionApi.DeletePartTimePosition)

		PartTiemPositionRouter.GET("getJobList", PartTimePositionApi.GetJobList)
		PartTiemPositionRouter.POST("createJob", PartTimePositionApi.CreateJob)
		PartTiemPositionRouter.POST("updateJob", PartTimePositionApi.UpdateJob)
		PartTiemPositionRouter.DELETE("deleteJob", PartTimePositionApi.DeleteJob)
	}
}
